rust/relay/src/server.rs

```diff
@@ -475,6 +430,54 @@ where
         self.pending_commands.pop_front()
     }
 
+    // TODO: It might be worth to do some caching here?
+    pub fn poll_timeout(&self) -> Option<SystemTime> {
```

Comment:
Wait, why are these `SystemTime` and not `std::time::Instant`?

`SystemTime` will slip or smear when NTP kicks in, and it's affected by leap seconds because it tracks UTC, I'm pretty sure.

Reply:
That is a good question. I remember having to change it for something in [`d2d8b09` (#1641)](https://github.com/firezone/firezone/pull/1641/commits/d2d8b09f0ffc795b4e98a097bb98f95823a31e8a) but I didn't add a comment why :man_facepalming: 

I can try changing it back. It might help with performance too as I've seen a fair chunk of CPU (single digit %) go into fetching the current time.

Reply:
Maybe there's a step in the protocol that requires both endpoints to have synced clocks to avoid a replay attack, I think TLS has something similar. It looks like this was added right after you switched to `SystemTime`: https://github.com/firezone/firezone/pull/1641/commits/04b34f5b0dd2c31bc44c69e1e3291abacc06a336#diff-93ca60c4df9a2b7f8e406f9bafe5090104c536c8a1e4327ae35a17ce1d1ea5a8R595

But that can ofc be done just for auth, with everything else using `Instant`

And I'm curious about the performance, too. I heard there's an optimization in Linux where libc can get the current time without a syscall, but I don't know exactly how it works.